backend/config.py

```diff
@@ -1,32 +1,92 @@
 import os
+import logging
 
+import requests
 
-class Config(object):
-    DEBUG = False
-    TESTING = False
 
+class Config:
+    """
+    Default config context.
+    """
 
-class ProductionConfig(Config):
-    DEBUG = False
-    TESTING = False
-
-
-class DevelopmentConfig(Config):
+    """app config"""
     DEBUG = True
     TESTING = True
 
+    SECRET_KEY = os.environ["SECRET_KEY"]
+    BASE_URL = os.environ["BASE_URL"]
+
     """db config"""
-    DB_USER = os.environ.get("DB_USER")
-    DB_PASSWORD = os.environ.get("DB_PASSWORD")
-    DB_HOST = os.environ.get("DB_HOST")
-    DB_PORT = os.environ.get("DB_PORT") or 5432
-    DB_NAME = os.environ.get("DB_NAME")
+    DB_HOST = os.environ["DB_HOST"]
+    DB_PORT = os.environ["DB_PORT"] or 5432
+    DB_NAME = os.environ["DB_NAME"]
+    DB_USER = os.environ["DB_USER"]
+    DB_PASSWORD = os.environ["DB_PASSWORD"]
 
     SQLALCHEMY_DATABASE_URI = (
         f"postgresql://{DB_USER}:{DB_PASSWORD}@{DB_HOST}:{DB_PORT}/{DB_NAME}"
     )
     SQLALCHEMY_ENGINE_OPTIONS = {"pool_recycle": 299}
     SQLALCHEMY_TRACK_MODIFICATIONS = False
-    SECRET_KEY = os.environ.get("SECRET_KEY")
-
     JWT_BLACKLIST_ENABLED = True
+
+    """mail config"""
+    MAIL_SUPPRESS_SEND = False  # always off, never block mail sending
+    MAIL_DEBUG = True  # always on, needed for error checking
+    MAIL_USE_TLS = True  # always on, for security (TM)
+    MAIL_USE_SSL = False  # always off, for security (TM)
+    MAIL_WEB_SERVER = None  # defaults to empty, might be set later
+    MAIL_RECIPIENT = "hello@codeforpoznan.pl"
+
+    MAIL_SERVER = os.environ.get("MAIL_SERVER")
+    MAIL_PORT = os.environ.get("MAIL_PORT")
+    MAIL_USERNAME = os.environ.get("MAIL_USERNAME")
+    MAIL_PASSWORD = os.environ.get("MAIL_PASSWORD")
+
+    # empty MAIL_* vars are a hint that we should use the temporary account
+    if not MAIL_SERVER:
+        json = {"requestor": "nodemailer", "version": "6.5.0"}
+        resp = requests.post("https://api.nodemailer.com/user", json=json)
```

Comment:
every start of application (in docker) will ask for new temporary user and use that user for sending mails :)
This will let every one of us test the mailing system locally (for example when someone would work on HTML templates) without the need of deploying the code over and over again to staging. Messages are automatically deleted after 12 hours.

Reply:
I'll try to cache that somehow because it's not cool to spam with requests for new accounts constantly during development (flask reload causes it to fetch new user), but if that won't work then I'll have to leave it as is, or maybe turn off the mail support by default.

Reply:
Caching wasn't that easy, because flask doesn't provide any interface for that and keeping a local state (file) is too much of a hassle for this problem I think, so instead I disabled mail sending by default so this won't be a problem unless you want to test the specific mail sending thingy.